pandas/indexes/multi.py

```diff
@@ -1007,6 +1008,9 @@ def from_tuples(cls, tuples, sortorder=None, names=None):
             # I think this is right? Not quite sure...
             raise TypeError('Cannot infer number of levels from empty list')
 
+        if not _check_equal_length(tuples):
```

Comment:
@shoyer makes a good point. checking cost is about the same as the actual construction (in this trivial example)

```
In [1]: tuples = [(1,2)]*1000000

In [2]: len(tuples[0])
Out[2]: 2

In [3]: %timeit any([len(x) != 2 for x in tuples])
10 loops, best of 3: 88.5 ms per loop

In [5]: %timeit pd.MultiIndex.from_tuples(tuples)
10 loops, best of 3: 108 ms per loop
```